### ISSUE TITLE
How to speed up video multiprocessing? 

### ISSUE THREAD

**Issue by anhdodotnet** [alexlaverty/ttsvibelounge#4]
I am wondering if my computer has 8 cores and 16 threads (16MB RAM, VGA Quando 1GB) is there any way to speed up the video processing it ?
![image](https://user-images.githubusercontent.com/25850343/196063878-642d0713-c940-4181-8f62-af5b99c924b7.png)

![image](https://user-images.githubusercontent.com/25850343/196063841-9229d905-2fb1-4c8e-a9aa-4a84b5457228.png)

I find the processing speed very slow:
![image](https://user-images.githubusercontent.com/25850343/196063935-6011fb39-38c7-4608-ae14-69f6923e7505.png)
CPU and GPU work at low levels, especially GPUs only work at 0-3%... that's an underperformance:
![image](https://user-images.githubusercontent.com/25850343/196064106-2d911a9f-0cf4-4083-bbcf-f8040b79cefc.png)

**Comment by alexlaverty** [comment]
I found an interesting thread about moviepy performance :
https://github.com/Zulko/moviepy/issues/645

I have implemented their suggestion :

```
post_video.write_videofile(v.filepath, threads=settings.threads, progress_bar = False)
```

I have defaulted it to 4 threads in the settings.py file, but if you have more cores than that you can try increasing the value and see if that improves performance for you

**Comment by anhdodotnet** [comment]
Thank you so much.

**Comment by beezisback** [comment]
thank you for the work done, I am deeply impressed.
I have some suggestions related to the project.
my discord is ambsd#8235

> I found an interesting thread about moviepy performance : [Zulko/moviepy#645](https://github.com/Zulko/moviepy/issues/645)
> 
> I have implemented their suggestion :
> 
> ```
> post_video.write_videofile(v.filepath, threads=settings.threads, progress_bar = False)
> ```
> 
> I have defaulted it to 4 threads in the settings.py file, but if you have more cores than that you can try increasing the value and see if that improves performance for you

**Comment by alexlaverty** [comment]
I've updated that line and tested it, it now runs without a progress bar as suggested in the thread, not sure how much of a performance improvement it will bring but worth a try. I notice in the thread they say that CompositeVideo clips can bog down performance, maybe there's too many layers in the composite clip, ie the background, the text, the overlay the narrator, etc.. maybe try turning off the overlay in the settings to remove a layer, the overlay is the snow effect you see over the text, might speed things up a bit.

**Comment by anhdodotnet** [comment]
> I've updated that line and tested it, it now runs without a progress bar as suggested in the thread, not sure how much of a performance improvement it will bring but worth a try. I notice in the thread they say that CompositeVideo clips can bog down performance, maybe there's too many layers in the composite clip, ie the background, the text, the overlay the narrator, etc.. maybe try turning off the overlay in the settings to remove a layer, the overlay is the snow effect you see over the text, might speed things up a bit.

Yes, I also tried removing "progress_bar = False" , it worked but didn't increase the speed.

**Comment by alexlaverty** [comment]
Just noticed this on stackoverflow about trying different codecs to see which one is fastest for your machine, you might want to play around with your codec :

https://stackoverflow.com/a/67815051
